### DISCUSSION TITLE
Inclusion/Exclusion stopped responding

### DISCUSSION THREAD

**Discussion by GrizzlyAK** [zwave-js/zwave-js#3363] [General]
Driver Ver: 8.2.1
Home Assistant: 2021.9.0
Running zwave-js under zwavejs2mqtt AddOn using websockets with the HA ZWave-JS Integration.

I recently added a device, and then later went to remove it, and the Remove Node dialog in the Home Assistant ZWave-JS Integration got stuck when I clicked the Start Exclusion button. I also noticed that the same non-responsiveness occurred when I clicked on the Start Inclusion button under Add Node. They would "light up", but I would never get the following dialog that provides progress, nor did anything happen with regard to adding or removing the node.

I restarted Home Assistant Core, but that did not resolve the issue. I had an outstanding HA OS upgrade (RPi) pending, so I did that, which required a restart of the device, and thus the zwave-js driver. Following this, the Add/Remove Node functions responded as expected. 

This was the first time I had encountered this behavior, requiring a restart, so wanted to report it, and see if anyone else has experienced it as well.

**Comment by GrizzlyAK** [comment]
OK, I don't restart my RPi very often, which is a benefit of using zwave-js, because it survives a restart of HA Core. Yay. However, this latest restart of zwave-js is FRIGHTENING. It took about an hour before I saw in the logs in the zwavejs2mqtt CP that the interviews were complete. However, I still have three _powered_ devices that are reporting as DEAD (there were more at one time). One is a Zooz ZEN21 in-wall light switch, one is a Leviton plug-in outlet, and one is a Zooz ZEN17 Universal Relay (powered). The Zen17 was alive, until I unplugged it, which then showed DEAD after several minutes. After plugging it back in, after about 45 minutes, it still reports DEAD. I just pinged it via the CP and it immediately came to life. Would it ultimately have come back alive on its own? That's important because these things control things. How long should it take for a powered device to come back to life? (I just pinged the remaining dead powered devicers and they immediately came back alive).

I also see in the Debug Window in the CP a LOT of INFO lines for the same device that are identical, e.g., motion sensors reporting a value 5 to 10 times in a row, not changes, but the exact same value. And things are extremely slow. Light switches turned on in HA take several seconds to respond. Sometimes the status in HA switches back and forth rapidly and ends up indicating the state opposite the actual device. I have 70 devices, 41 are powered, 29 are battery with a 2 hour wakeup on all battery devices.

One other thing I've noticed with my Zooz ZSE40 multisensors (which I noticed back with OZW 1.4) is that during the interview process, the motion sensor is triggered MULTIPLE times. Not sure why, but this wreaks havoc with sensors that are tied to the device, like garage lights. Is this normal, or perhaps related to this particular sensor's FW in how it responds to interviews? In fact, the interview process is "complete", yet these devices just keep reporting motion when I know there is none. Under OZW 1.4, it only reported once, maybe twice on restart. But this just keeps triggering automations again and again. Not sure what is going on.

I guess I'm just trying to figure out if this is _normal_ behavior following a restart.

**Reply by AlCalzone** [reply]
> I guess I'm just trying to figure out if this is normal behavior following a restart.

Definitely not, but this is likely an erratic device.

> LOT of INFO lines for the same device that are identical, e.g., motion sensors reporting a value 5 to 10 times in a row, not changes, but the exact same value

Interestingly, I just reviewed another logfile where a device kept flooding and DDoS'ing the entire network for minutes. This also seems to have been a Zooz device, which went into full-on panic mode because it couldn't reach the controller:
https://github.com/zwave-js/node-zwave-js/issues/3359#issuecomment-922471388
Without seeing actual driver logs for your issue, I suspect something like this is happening too.

> After plugging it back in, after about 45 minutes, it still reports DEAD.

The status is not updated unless there was communication between the controller and the device. If the device didn't tell us that it is back alive, we can't know.

**Reply by GrizzlyAK** [reply]
I understand that last sentence about the ZEN17, but is it normal for a powered device to not announce itself when it is turned on? Is this something I should take up with Zooz? Perhaps it was just the delay or the other issue that was causing this.

Also, it appeared to be multiple devices that were flooding the logs with identical INFO lines. HOWEVER...

I finally got fed up and just shut down and powered off my RPi completely for about 10 minutes, and when it restarted, everything came back up _perfectly_, with everything responding quickly within minutes, and NO false positives (that I saw) from the ZSE40 motion sensors. Something weird happened in the HA Core upgrade/restart, OS upgrade/restart (in that order) to zwave-js, zwavejs2mqtt, or the HA zwavejs integration. I don't think that all my devices suddenly decided to revolt ;-)

Thanks for your response. It is still worrisome, though, but glad I'm back to normal.

**Reply by GrizzlyAK** [reply]
This is odd. I just tried to open a gzip of the zwavejs driver logs in the zwavejs2mqtt dashboard, and none of them can be opened, or are corrupt. I'll check tomorrow after today's is retired to see if perhaps it was the glich that caused it, or if I still have issues. Not sure who writes the gzip backups, if the driver or zwavejs2mqtt. ?

**Reply by kpine** [reply]
> This is odd. I just tried to open a gzip of the zwavejs driver logs in the zwavejs2mqtt dashboard, and none of them can be opened, or are corrupt

It looks like you've discovered a zwavejs2mqtt bug. I just tried this myself; downloading gzip files from the Store Explorer results in 0-byte files. They are fine on the server's filesystem.

To properly download the gzip files, click in the checkbox **next** to the gzip file, not the gzip file itself. When you click the check box you'll get a cog icon on the bottom right. Click that and you'll get an option to download a zip file (green icon). The zip file will contain any of the files that were selected, even if just one file. This process is also [shown](https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/bug_report?id=download-zip) in the troubleshooting docs, with a slightly outdated screen shot.

**Reply by GrizzlyAK** [reply]
Thanks @kpine, that worked perfectly. I think in the past I've done this the way you suggested, because I recognized the dialogs, etc. Not sure why I clicked on the file itself this time, but saw the Download option and that's what didn't work. I'll let zwavejs2mqtt folks know. Cheers!

**Reply by AlCalzone** [reply]
> but is it normal for a powered device to not announce itself when it is turned on?

There's nothing in the specs saying that it should. I also have a wall plug that needs to be operated manually before it announces itself.

>  I don't think that all my devices suddenly decided to revolt ;-)

One can be enough to bring down the network. Maybe all you saw was the driver not being able to send commands and therefore not being able to determine the other nodes' statuses.

**Reply by GrizzlyAK** [reply]
> There's nothing in the specs saying that it should.

Good to know.

BTW, just wanted to say that I'm very, VERY pleased with the way zwave-js works. Since upgrading a month or so ago, it has performed flawlessly and is so much faster than OZW 1.4. That's why this event was so _weird_! Thank you for all you do in making this a reality and keeping or worlds humming along.

**Reply by GrizzlyAK** [reply]
Hi Al. I had a power outage today. My RPi is on UPS, so I shut it down properly. When power was restored, I booted my Pi and the system came up, but I noticed right away that most (if not all) of my powered devices were "unavailable" in HA UI like last time.  Also, the entities (like in-wall light switches (ZEN21s) were unresponsive as well). Basically, all these devices just didn't work. Since Zwave was restarted, I waited a bit to see if things would come online, but nothing changed. I didn't check the logs this time. Like you said last time, the specs don't state that a powered device has to check in on startup, but it would seem that it would be necessary to complete the interview process. It is very similar to the behavior I reported after restarting last time following an upgrade. This was just a restart, nothing changed before shutdown. 

After a while, I rebooted my Pi again, and when it came back up, everything was normal, with all powered devices online. I'm not sure what is going on, or where the issue may lie, but it is unnerving given that some of my powered devices are pretty important, like heat. I rely on this system to reset properly when away in event of power outage. It seems something is getting stuck somewhere.

Any thoughts on why this seems to be a recurring issue for me, or suggestions on what I might do to track down why this is happening? I don't really know if this is ZWave-JS server, ZWavejs2mqtt (doubtful), or the HA ZWave Integration, so I'm sorry if this is the wrong place to bring this up.

**Reply by AlCalzone** [reply]
Without seeing logs, my money is on the stick itself. According to the specs, the driver should soft-reset (reboot) the stick on startup, but that doesn't happen currently. So what you're experiencing is likely an unresponsive stick. You should be able to see that by looking at the statistics (which are a bit hidden):
![grafik](https://user-images.githubusercontent.com/17641229/134801462-e0c702be-9bec-41af-b11a-c0de397820be.png)
In your case I'd expect a bunch of timeouts.

I have an [open PR](https://github.com/zwave-js/node-zwave-js/pull/3262) to make use of the softreset, but the fact that almost everyone is running Docker makes this nearly impossible to use. Unless the host is configured properly, the stick will be disconnected until the next container reboot (which would cause zwave-js to reset it again).
And apparently some other sticks go completely dark a few seconds after reset 🤷🏻‍♂️

So the only way to reset the stick now is:
* shut down zwavejs2mqtt
* pull stick
* put it back
* start zwavejs2mqtt again
